### DISCUSSION TITLE
Scenes Wizard exactly-once message handling issue

### DISCUSSION THREAD

**Discussion by dmrz** [aiogram/aiogram#1385] [Q&A] [answered]
Hi! I've got this issue where it's possible to end up in scene's exit handler more than once, when you send messages fast enough to the bot (say by eagerly clicking markup keyboard button frequently). This is also reproducible with Quiz example from [documentation](https://docs.aiogram.dev/en/latest/dispatcher/finite_state_machine/scene.html). It seems like previous scene instance might handle another incoming message while being in a leaving state (if I understand it correctly) and this can possibly be the reason. I totally understand that this feature is experimental, but maybe there's a known workaround for this issue already? There is no such issue If I implement the same functionality using StatesGroup based handlers, but I like Scenes Wizard for much cleaner codebase.

**Comment by andrew000** [comment, marked as answer]
Do you use EventIsolation?

**Reply by dmrz** [reply]
I didn't. SimpleEventIsolation resolves the issue. Thanks a lot! I have neglected `create_dispatcher` part from documentation and missed that point 🫠
